document-render-blocking.md

```diff
@@ -36,16 +75,104 @@ The timing for when this attribute can block rendering for a Document is already
 <html blocking="render">
+<script>
+  // The set of element IDs that should block rendering.
+  let blockingElementIds = new Set();
+
+  function maybeUnblockRendering() {
+    if (blockingElementIds.size == 0) {
+      document.documentElement.blocking="";
+    }
+  }
+
+  // The value returned by getState() is set by the old Document in
+  // the `navigate` event. It tracks whether the old Document add a
+  // `view-transition-name` to the header.
+  if (navigation.initialLoad.from().getState().morphHeader) {
+    blockingElementIds.add("header-id");
+  }
+  maybeUnblockRendering();
+</script>
 <body>
 …
+<div id="header-id">
+  ...
+</div>
 <script>
-   if (doPartialBlocking) {
-      document.documentElement.blocking="";
-   }
+  // When an element is parsed, remove it from the blocking set and
+  // unblock rendering if all blocking elements have been parsed.
+  blockingElementIds.delete("header-id");
+  maybeUnblockRendering();
+</script>
+</body>
+</html>
+```
+
+## Meta Tag with Blocking ElementIds
+
+Add a new meta tag with the name `blocking-elements` and `content` attribute set to the list of [element IDs](https://dom.spec.whatwg.org/#concept-id) which must be parsed before rendering. `*` is a special keyword which implies every ID should be blocking.
+
+Each Document has a render-blocking-until-parsed element ids set, initially empty. A Document is [render-blocked](https://html.spec.whatwg.org/#render-blocked) if this set is non-empty.
+
+- If the value of the `content` attribute changes, and the Document [allows adding render blocking elements](https://html.spec.whatwg.org/#allows-adding-render-blocking-elements), the render-blocking-until-parsed element ids is set to the new attribute value.
+   This means authors can run script to configure the list until the body tag is parsed (after which no new render blocking resources can be added).
+
+- If the value of the `content` attribute changes, and the Document **does not** [allow adding render blocking elements](https://html.spec.whatwg.org/#allows-adding-render-blocking-elements), render-blocking-until-parsed element ids is set to be an intersection of the existing value and the new attribute value.
+   This means authors can run script to remove render-blocking elements after the body tag is parsed but can't add more elements. This allows authors to implement their own timeout if needed.
+
+- Each time an element's ID value changes, the browser checks if the set of elements which have been completely parsed (i.e. the end tag has been parsed) include all IDs in the render-blocking-until-parsed element ids set. If yes, the render-blocking-until-parsed element ids set is cleared.
+
+The pro of this approach is that its easier to block on a specific set of elements, which makes it more likely that authors will consider partial blocking. The con is new syntax which requires defining subtle interactions (like script changing element IDs after parsing).
+
+### Including Blocking Tokens
```

Comment:
I'd also note that a typo in this list would cause the parser to block on the whole document (waiting for the non-existent id), but in that case we can throw an error to the console

Reply:
Done.